### ISSUE TITLE
VLAN breaks Ethernet Broadband

### ISSUE THREAD

**Issue by seud0nym** [seud0nym/tch-gui-unhide#156]
Hi @seud0nym,

Sorry to hijack this issue, looks like I also have a problem when trying to set up VLANs.

I've just tried running VLANs from the Local Network screen however it disconnects my WAN connection straight away and switches over to the mobile backup. I have a PPPoE connection for my WAN

I also noticed that in the broadband card it shows "VLAN 1" as active. If I go into that card the VLAN settings have been switched off so it must be pulling it from the local network card (might be a clue to follow)

If I switch VLANs off it reconnects straight away 

Big fan of you and what you've done so far with these modems, it's awesome! 

Thanks again for your help

Firmware Version: 20.3.c.0432-MR21.1-RA

<img width="1552" alt="Screen Shot 2023-06-02 at 5 37 37 pm" src="https://github.com/seud0nym/tch-gui-unhide/assets/22275247/59453450-2df2-4cf6-81cc-239c02871d7a">

<img width="1552" alt="Screen Shot 2023-06-02 at 5 37 27 pm" src="https://github.com/seud0nym/tch-gui-unhide/assets/22275247/37f287b9-b0a2-47b6-9750-843f22fc8b20">

_Originally posted by @gingerninja234 in https://github.com/seud0nym/tch-gui-unhide/issues/152#issuecomment-1573296009_

**Comment by seud0nym** [comment]
Can you please post the output of the following commands (when you have VLANs enabled):
```
transformer-cli get uci.network.interface.@lan. | sort
transformer-cli get uci.network.interface.@wan. | sort
transformer-cli get uci.network.device. | sort
```

**Comment by seud0nym** [comment]
I forgot a command. What is the output from:
```
mdkshell vlan show
```

**Comment by gingerninja234** [comment]
Sure, here are the results

**transformer-cli get uci.network.interface.@lan. | sort**

`root@DJA0231:~# transformer-cli get uci.network.interface.@lan. | sort
uci.network.interface.@lan.ac [string] = 
uci.network.interface.@lan.accept_ra [string] = 
uci.network.interface.@lan.adv_interface [string] = 
uci.network.interface.@lan.adv_preferred_lifetime [string] = 
uci.network.interface.@lan.adv_subnet [string] = 
uci.network.interface.@lan.adv_valid_lifetime [string] = 
uci.network.interface.@lan.apn [string] = 
uci.network.interface.@lan.atmdev [string] = 
uci.network.interface.@lan.authfail [string] = 
uci.network.interface.@lan.auto [string] = 
uci.network.interface.@lan.bridge_flag [string] = 
uci.network.interface.@lan.broadcast [string] = 
uci.network.interface.@lan.buffering [string] = 
uci.network.interface.@lan.clientid [string] = 
uci.network.interface.@lan.connect [string] = 
uci.network.interface.@lan.connect_time [string] = 
uci.network.interface.@lan.customroutes [string] = 
uci.network.interface.@lan.dadtransmits [string] = 
uci.network.interface.@lan.defaultroute [string] = 
uci.network.interface.@lan.delegate [string] = 
uci.network.interface.@lan.demand [string] = 
uci.network.interface.@lan.dev_desc [string] = 
uci.network.interface.@lan.device [string] = 
uci.network.interface.@lan.df [string] = 
uci.network.interface.@lan.disconnect [string] = 
uci.network.interface.@lan.dnsset [string] = 
uci.network.interface.@lan.domain [string] = 
uci.network.interface.@lan.enabled [string] = 
uci.network.interface.@lan.encaps [string] = 
uci.network.interface.@lan.force_link [string] = 1
uci.network.interface.@lan.forceprefix [string] = 
uci.network.interface.@lan.gateway [string] = 
uci.network.interface.@lan.graceful_restart [string] = 
uci.network.interface.@lan.hostname [string] = 
uci.network.interface.@lan.icsum [string] = 
uci.network.interface.@lan.iface [string] = 
uci.network.interface.@lan.iface6rd [string] = 
uci.network.interface.@lan.iface_464xlat [string] = 
uci.network.interface.@lan.ifname [string] = vlan_eth0_1 vlan_eth1_1 vlan_eth2_1 vlan_eth3_1
uci.network.interface.@lan.igmpversion [string] = 
uci.network.interface.@lan.ikey [string] = 
uci.network.interface.@lan.ip4prefixlen [string] = 
uci.network.interface.@lan.ip4table [string] = 
uci.network.interface.@lan.ip6addr [string] = 
uci.network.interface.@lan.ip6assign [string] = 0
uci.network.interface.@lan.ip6gw [string] = 
uci.network.interface.@lan.ip6hint [string] = 0
uci.network.interface.@lan.ip6ifaceid [string] = 
uci.network.interface.@lan.ip6prefix [string] = 
uci.network.interface.@lan.ip6prefixlen [string] = 
uci.network.interface.@lan.ip6table [string] = 
uci.network.interface.@lan.ipaddr [string] = 10.0.20.1
uci.network.interface.@lan.ipv6 [string] = 
uci.network.interface.@lan.iseqno [string] = 
uci.network.interface.@lan.keepalive [string] = 
uci.network.interface.@lan.lease [string] = 
uci.network.interface.@lan.lease_expires [string] = 
uci.network.interface.@lan.macaddr [string] = 
uci.network.interface.@lan.maxwait [string] = 
uci.network.interface.@lan.metric [string] = 
uci.network.interface.@lan.mtu [string] = 
uci.network.interface.@lan.name [string] = 
uci.network.interface.@lan.neighgcstaletime [string] = 
uci.network.interface.@lan.neighreachabletime [string] = 
uci.network.interface.@lan.netmask [string] = 255.255.255.0
uci.network.interface.@lan.network [string] = 
uci.network.interface.@lan.norelease [string] = 
uci.network.interface.@lan.noslaaconly [string] = 
uci.network.interface.@lan.ocsum [string] = 
uci.network.interface.@lan.okey [string] = 
uci.network.interface.@lan.oseqno [string] = 
uci.network.interface.@lan.password [password] = 
uci.network.interface.@lan.pdnid [string] = 
uci.network.interface.@lan.peer6addr [string] = 
uci.network.interface.@lan.peeraddr [string] = 
uci.network.interface.@lan.peerdns [string] = 
uci.network.interface.@lan.pincode [string] = 
uci.network.interface.@lan.pppd_options [string] = 
uci.network.interface.@lan.pppoerelay.@1.value [string] = 
uci.network.interface.@lan.profile [string] = 
uci.network.interface.@lan.proto [string] = static
uci.network.interface.@lan.reqaddress [string] = 
uci.network.interface.@lan.reqopts [string] = 
uci.network.interface.@lan.reqprefix [string] = 
uci.network.interface.@lan.rpfilter [string] = 1
uci.network.interface.@lan.send_rs [string] = 
uci.network.interface.@lan.sendopts [string] = 
uci.network.interface.@lan.server [string] = 
uci.network.interface.@lan.service [string] = 
uci.network.interface.@lan.session_id [string] = 
uci.network.interface.@lan.sourcerouting [string] = 
uci.network.interface.@lan.srcintf [string] = 
uci.network.interface.@lan.stp [string] = 
uci.network.interface.@lan.tch_ip6assign [string] = 
uci.network.interface.@lan.ttl [string] = 
uci.network.interface.@lan.tunlink [string] = 
uci.network.interface.@lan.tunnelid [string] = 
uci.network.interface.@lan.type [string] = bridge
uci.network.interface.@lan.unnumbered [string] = 
uci.network.interface.@lan.updatekey [string] = 
uci.network.interface.@lan.username [string] = 
uci.network.interface.@lan.vci [string] = 
uci.network.interface.@lan.vendorclass [string] = 
uci.network.interface.@lan.vendorid [string] = 
uci.network.interface.@lan.vpi [string] = 
uci.network.interface.@lan.wan_device [string] = 
uci.network.interface.@lan.weakif [string] = 
uci.network.interface.@lan.zone [string] = `

**transformer-cli get uci.network.interface.@wan. | sort** 

`root@DJA0231:~# transformer-cli get uci.network.interface.@wan. | sort
uci.network.interface.@wan.ac [string] = 
uci.network.interface.@wan.accept_ra [string] = 
uci.network.interface.@wan.adv_interface [string] = 
uci.network.interface.@wan.adv_preferred_lifetime [string] = 
uci.network.interface.@wan.adv_subnet [string] = 
uci.network.interface.@wan.adv_valid_lifetime [string] = 
uci.network.interface.@wan.apn [string] = 
uci.network.interface.@wan.atmdev [string] = 
uci.network.interface.@wan.authfail [string] = 
uci.network.interface.@wan.auto [string] = 1
uci.network.interface.@wan.bridge_flag [string] = 
uci.network.interface.@wan.broadcast [string] = 
uci.network.interface.@wan.buffering [string] = 
uci.network.interface.@wan.clientid [string] = 
uci.network.interface.@wan.connect [string] = 
uci.network.interface.@wan.connect_time [string] = 
uci.network.interface.@wan.customroutes [string] = 
uci.network.interface.@wan.dadtransmits [string] = 
uci.network.interface.@wan.defaultroute [string] = 
uci.network.interface.@wan.delegate [string] = 
uci.network.interface.@wan.demand [string] = 
uci.network.interface.@wan.dev_desc [string] = 
uci.network.interface.@wan.device [string] = 
uci.network.interface.@wan.df [string] = 
uci.network.interface.@wan.disconnect [string] = 
uci.network.interface.@wan.dnsset [string] = 
uci.network.interface.@wan.domain [string] = 
uci.network.interface.@wan.enabled [string] = 
uci.network.interface.@wan.encaps [string] = 
uci.network.interface.@wan.force_link [string] = 
uci.network.interface.@wan.forceprefix [string] = 
uci.network.interface.@wan.gateway [string] = 
uci.network.interface.@wan.graceful_restart [string] = 1
uci.network.interface.@wan.hostname [string] = 
uci.network.interface.@wan.icsum [string] = 
uci.network.interface.@wan.iface [string] = 
uci.network.interface.@wan.iface6rd [string] = 0
uci.network.interface.@wan.iface_464xlat [string] = 
uci.network.interface.@wan.ifname [string] = vlan_eth4_1
uci.network.interface.@wan.igmpversion [string] = 
uci.network.interface.@wan.ikey [string] = 
uci.network.interface.@wan.ip4prefixlen [string] = 
uci.network.interface.@wan.ip4table [string] = 
uci.network.interface.@wan.ip6addr [string] = 
uci.network.interface.@wan.ip6assign [string] = 
uci.network.interface.@wan.ip6gw [string] = 
uci.network.interface.@wan.ip6hint [string] = 
uci.network.interface.@wan.ip6ifaceid [string] = 
uci.network.interface.@wan.ip6prefix [string] = 
uci.network.interface.@wan.ip6prefixlen [string] = 
uci.network.interface.@wan.ip6table [string] = 
uci.network.interface.@wan.ipaddr [string] = 
uci.network.interface.@wan.ipv6 [string] = 
uci.network.interface.@wan.iseqno [string] = 
uci.network.interface.@wan.keepalive [string] = 4,20
uci.network.interface.@wan.lease [string] = 
uci.network.interface.@wan.lease_expires [string] = 
uci.network.interface.@wan.macaddr [string] = 
uci.network.interface.@wan.maxwait [string] = 
uci.network.interface.@wan.metric [string] = 10
uci.network.interface.@wan.mtu [string] = 
uci.network.interface.@wan.name [string] = 
uci.network.interface.@wan.neighgcstaletime [string] = 
uci.network.interface.@wan.neighreachabletime [string] = 
uci.network.interface.@wan.netmask [string] = 
uci.network.interface.@wan.network [string] = 
uci.network.interface.@wan.norelease [string] = 
uci.network.interface.@wan.noslaaconly [string] = 
uci.network.interface.@wan.ocsum [string] = 
uci.network.interface.@wan.okey [string] = 
uci.network.interface.@wan.oseqno [string] = 
uci.network.interface.@wan.password [password] = new2dsl
uci.network.interface.@wan.pdnid [string] = 
uci.network.interface.@wan.peer6addr [string] = 
uci.network.interface.@wan.peeraddr [string] = 
uci.network.interface.@wan.peerdns [string] = 
uci.network.interface.@wan.pincode [string] = 
uci.network.interface.@wan.pppd_options [string] = 
uci.network.interface.@wan.profile [string] = 
uci.network.interface.@wan.proto [string] = pppoe
uci.network.interface.@wan.reqaddress [string] = 
uci.network.interface.@wan.reqopts [string] = 
uci.network.interface.@wan.reqprefix [string] = 
uci.network.interface.@wan.rpfilter [string] = 
uci.network.interface.@wan.send_rs [string] = 
uci.network.interface.@wan.sendopts [string] = 
uci.network.interface.@wan.server [string] = 
uci.network.interface.@wan.service [string] = 
uci.network.interface.@wan.session_id [string] = 
uci.network.interface.@wan.sourcerouting [string] = 
uci.network.interface.@wan.srcintf [string] = 
uci.network.interface.@wan.stp [string] = 
uci.network.interface.@wan.tch_ip6assign [string] = 
uci.network.interface.@wan.ttl [string] = 
uci.network.interface.@wan.tunlink [string] = 
uci.network.interface.@wan.tunnelid [string] = 
uci.network.interface.@wan.type [string] = 
uci.network.interface.@wan.unnumbered [string] = 
uci.network.interface.@wan.updatekey [string] = 
uci.network.interface.@wan.username [string] = newdsluser@bigpond.com
uci.network.interface.@wan.vci [string] = 
uci.network.interface.@wan.vendorclass [string] = 
uci.network.interface.@wan.vendorid [string] = 
uci.network.interface.@wan.vpi [string] = 
uci.network.interface.@wan.wan_device [string] = 
uci.network.interface.@wan.weakif [string] = 
uci.network.interface.@wan.zone [string] = `

**transformer-cli get uci.network.device. | sort**
`root@DJA0231:~# transformer-cli get uci.network.device. | sort
uci.network.device.@atm_8_35.enabled [string] = 
uci.network.device.@atm_8_35.ifname [string] = 
uci.network.device.@atm_8_35.igmpversion [string] = 
uci.network.device.@atm_8_35.ipv6 [string] = 
uci.network.device.@atm_8_35.macaddr [string] = 
uci.network.device.@atm_8_35.mtu [string] = 
uci.network.device.@atm_8_35.name [string] = atm_8_35
uci.network.device.@atm_8_35.type [string] = 
uci.network.device.@atm_8_35.vid [string] = 
uci.network.device.@eth4.enabled [string] = 
uci.network.device.@eth4.ifname [string] = 
uci.network.device.@eth4.igmpversion [string] = 
uci.network.device.@eth4.ipv6 [string] = 
uci.network.device.@eth4.macaddr [string] = 
uci.network.device.@eth4.mtu [string] = 1500
uci.network.device.@eth4.name [string] = eth4
uci.network.device.@eth4.type [string] = 
uci.network.device.@eth4.vid [string] = 
uci.network.device.@ptm0.enabled [string] = 
uci.network.device.@ptm0.ifname [string] = 
uci.network.device.@ptm0.igmpversion [string] = 
uci.network.device.@ptm0.ipv6 [string] = 
uci.network.device.@ptm0.macaddr [string] = 
uci.network.device.@ptm0.mtu [string] = 
uci.network.device.@ptm0.name [string] = ptm0
uci.network.device.@ptm0.type [string] = 
uci.network.device.@ptm0.vid [string] = 
uci.network.device.@vlan_eth0_1.enabled [string] = 
uci.network.device.@vlan_eth0_1.ifname [string] = eth0
uci.network.device.@vlan_eth0_1.igmpversion [string] = 
uci.network.device.@vlan_eth0_1.ipv6 [string] = 
uci.network.device.@vlan_eth0_1.macaddr [string] = 
uci.network.device.@vlan_eth0_1.mtu [string] = 
uci.network.device.@vlan_eth0_1.name [string] = vlan_eth0_1
uci.network.device.@vlan_eth0_1.type [string] = 8021q
uci.network.device.@vlan_eth0_1.vid [string] = 1
uci.network.device.@vlan_eth1_1.enabled [string] = 
uci.network.device.@vlan_eth1_1.ifname [string] = eth1
uci.network.device.@vlan_eth1_1.igmpversion [string] = 
uci.network.device.@vlan_eth1_1.ipv6 [string] = 
uci.network.device.@vlan_eth1_1.macaddr [string] = 
uci.network.device.@vlan_eth1_1.mtu [string] = 
uci.network.device.@vlan_eth1_1.name [string] = vlan_eth1_1
uci.network.device.@vlan_eth1_1.type [string] = 8021q
uci.network.device.@vlan_eth1_1.vid [string] = 1
uci.network.device.@vlan_eth2_1.enabled [string] = 
uci.network.device.@vlan_eth2_1.ifname [string] = eth2
uci.network.device.@vlan_eth2_1.igmpversion [string] = 
uci.network.device.@vlan_eth2_1.ipv6 [string] = 
uci.network.device.@vlan_eth2_1.macaddr [string] = 
uci.network.device.@vlan_eth2_1.mtu [string] = 
uci.network.device.@vlan_eth2_1.name [string] = vlan_eth2_1
uci.network.device.@vlan_eth2_1.type [string] = 8021q
uci.network.device.@vlan_eth2_1.vid [string] = 1
uci.network.device.@vlan_eth3_1.enabled [string] = 
uci.network.device.@vlan_eth3_1.ifname [string] = eth3
uci.network.device.@vlan_eth3_1.igmpversion [string] = 
uci.network.device.@vlan_eth3_1.ipv6 [string] = 
uci.network.device.@vlan_eth3_1.macaddr [string] = 
uci.network.device.@vlan_eth3_1.mtu [string] = 
uci.network.device.@vlan_eth3_1.name [string] = vlan_eth3_1
uci.network.device.@vlan_eth3_1.type [string] = 8021q
uci.network.device.@vlan_eth3_1.vid [string] = 1
uci.network.device.@vlan_eth4_1.enabled [string] = 
uci.network.device.@vlan_eth4_1.ifname [string] = eth4
uci.network.device.@vlan_eth4_1.igmpversion [string] = 
uci.network.device.@vlan_eth4_1.ipv6 [string] = 
uci.network.device.@vlan_eth4_1.macaddr [string] = 
uci.network.device.@vlan_eth4_1.mtu [string] = 
uci.network.device.@vlan_eth4_1.name [string] = vlan_eth4_1
uci.network.device.@vlan_eth4_1.type [string] = 8021q
uci.network.device.@vlan_eth4_1.vid [string] = 1`

**mdkshell vlan show**
`root@DJA0231:~# mdkshell vlan show
VLAN 1 ports: 0-3 8   untagged: 0-3 `


Looks like it has deleted my PPoE username and password (from the output of  uci.network.interface.@wan.). Maybe if I enter them it might work again

**Comment by seud0nym** [comment]
That is very helpful.  This is the problem:
```
uci.network.interface.@wan.ifname [string] = vlan_eth4_1 
```
It should be `eth4` not `vlan_eth4_1 `, so it looks like I have incorrectly updated the WAN interface, instead of limiting my updated to LAN interfaces. Unfortunately, it will probably do that every time you enable or change VLANs, so best to turn it off at the moment. That will revert it back to `eth4`.

**Comment by gingerninja234** [comment]
Awesome! I see, I had a feeling the interface may have got renamed

Will keep VLANs off for now

**Comment by seud0nym** [comment]
I have committed the fix. If you want to try it, you can do update with this command:
```
./tch-gui-unhide -u && ./tch-gui-unhide -y
```

**Comment by gingerninja234** [comment]
It works! Thank you for the quick fix! :)

**Comment by seud0nym** [comment]
Fixed in stable release 2023.06.23
